glocaltokens/client.py

```diff
@@ -365,13 +371,29 @@ def get_google_devices(
 
         devices: list[Device] = []
 
+        def is_dict_with_valid_ipv4_addresses(data: dict[str, str]) -> bool:
+            # Test if the data structure is correct and if each entry contains a
+            # valid IPv4 address.
+            return isinstance(data, dict) and all(
+                isinstance(x, str) and is_valid_ipv4_address(x) for x in data.values()
+            )
+
+        if addresses and not is_dict_with_valid_ipv4_addresses(addresses):
```

Comment:
Tbh i don't know if there is any benefit of the proposed approach but it looks more readable, isn't it? Or is it just me? :D 

```suggestion
        if addresses and is_dict_with_valid_ipv4_addresses(addresses) is False:
```

Reply:
Your way is more readable, but the other way is the pythonic way to do it, as far as i know

Reply:
haha same :D